Makefile

```diff
@@ -282,16 +290,82 @@ docker-cilium-image-for-developers:
 	     --build-arg LIBNETWORK_PLUGIN=\
 	     -t "$(DOCKER_DEV_ACCOUNT)/cilium-dev:latest" . -f ./cilium-dev.Dockerfile
 
-docker-image: clean docker-image-no-clean docker-operator-image docker-plugin-image docker-hubble-relay-image
+docker-image: docker-image-no-clean docker-operator-image docker-plugin-image docker-hubble-relay-image
 
-docker-image-no-clean: GIT_VERSION
-	$(QUIET)$(CONTAINER_ENGINE) build \
+# Require user to have defined DOCKER_BUILDKIT (to any value) to use it.
+BUILDKIT_DOCKERFILE_FILTER =
+ifdef DOCKER_BUILDKIT
+	# Export with value expected by docker
+	export DOCKER_BUILDKIT=1
+	BUILDKIT_DOCKERFILE_FILTER := | sed -e "1s|^\#.*|\# syntax = docker/dockerfile:experimental|" -e "s|^RUN\(.*\)make|RUN --mount=type=cache,target=/root/.cache/go-build\1make|"
```

Comment:
I would much rather see this added permanently, what version of Docker supports this option?

Reply:
I spoke with @nebril and it sounds like upgrading Docker in Jenkins wouldn't be a big deal :)

Reply:
dev VM has docker 19.03 and it still has a bug where it sometimes fails to pull the base image on the FROM line, behaving as if it was pulled, but then "make" or "/bin/sh" can't be found (see @joestringer's comments below). At the least, enabling DOCKER_BUILDKIT in the CI takes some testing in addition updates (if needed).